spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/VectorizedSparkOrcReaders.java

```diff
@@ -394,8 +394,12 @@ public ColumnVector convert(
         @Override
         public ColumnarArray getArray(int rowId) {
           int index = getRowIndex(rowId);
-          return new ColumnarArray(
-              elementVector, (int) listVector.offsets[index], (int) listVector.lengths[index]);
+          if (listVector.noNulls || !listVector.isNull[index]) {
+            return new ColumnarArray(
+                elementVector, (int) listVector.offsets[index], (int) listVector.lengths[index]);
+          }
+
+          return null;
```

Comment:
it would be good to have a test that actually exercises this path and reaches the `null` here (same for `getMap` below). I checked locally and no tests in `TestSparkOrcReader` actually reach this